kernel/src/net/socket/netlink/route/kernel/util.rs

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use crate::{
+    net::socket::netlink::route::message::{
+        CMessageSegmentHeader, DoneSegment, MsgSegment, SegmentHeaderCommonFlags,
+    },
+    prelude::*,
+};
+
+/// Appends a done segment as the last segment of the provided segments, if necessary.
+///
+/// A done segment will be added if:
+/// 1. `segments.len()` > 1, as mulitible segments must be terminated with a done segment.
+/// 2. (TODO) The request_header's flags contain the `ACK` flag, explicitly requesting a done segment.
+pub fn append_done_segment(request_header: &CMessageSegmentHeader, segments: &mut Vec<MsgSegment>) {
+    // TODO: Deal with the `NetlinkMessageCommonFlags::ACK` flag
+    if segments.len() <= 1 {
+        return;
+    }
+
+    let done_segment = DoneSegment::new_from_request(request_header, None);
+    segments.push(MsgSegment::Done(done_segment));
+}
+
+/// Adds the `MULTI` flag to all segments in `segments` if `segments.len() > 1`.
+pub fn add_multi_flag(segments: &mut Vec<MsgSegment>) {
```

Comment:
I think we should merge `append_done_segment` and `add_multi_flag` to one (public) method (internally it can use two private methods to implement the functionality), which seems to be more clear.